pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -209,37 +209,37 @@ func GetParam(kmsConfig map[string]string, param string) string {
 }
 
 // ValidateConnectionDetails validates mandatory KMS connection details
-func ValidateConnectionDetails(ctx context.Context, clusterdContext *clusterd.Context, securitySpec *cephv1.SecuritySpec, ns string) error {
+func ValidateConnectionDetails(ctx context.Context, clusterdContext *clusterd.Context, kms *cephv1.KeyManagementServiceSpec, ns string) error {
 	// Lookup mandatory connection details
 	for _, config := range kmsMandatoryConnectionDetails {
-		if GetParam(securitySpec.KeyManagementService.ConnectionDetails, config) == "" {
+		if GetParam(kms.ConnectionDetails, config) == "" {
 			return errors.Errorf("failed to validate kms config %q. cannot be empty", config)
```

Comment:
The function is to check the connection with kms backend more likely to have kms struct than the security struct., IMO the package `kms` kept `osd` make it bit confusing, it mostly contains generic apis for kms connections, so it may kept different location than under `osd`. Will address this in a different PR.